.github/workflows/gradle-build.yml

```diff
@@ -0,0 +1,61 @@
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
+
+on:
+  pull_request:
+    branches: [ dev ]
+    paths:
+      - 'src/**'
+      - '.github/**'
+      - '!.gradle/wrapper'
+      - '!.gitignore'
+      - '!LICENSE'
+      - '!THIRD PARTY NOTICES'
+      - '!*.md'
+      - '*.gradle'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: '16'
+          distribution: 'adopt'
+          cache: gradle
+      - name: Detect Secrets
+        uses: RobertFischer/detect-secrets-action@v2.0.0
+      - run: .\scripts\decodeAndWrite.ps1 -encodedValue $env:ENCODED_VALUE -outputPath $env:OUTPUT_PATH
+        shell: pwsh
+        env:
+          ENCODED_VALUE: ${{ secrets.LOCAL_PROPERTIES }}
+          OUPUT_PATH: .\local.properties
+      - run: .\scripts\decodeAndWrite.ps1 -encodedValue $env:ENCODED_VALUE -outputPath $env:OUTPUT_PATH
+        shell: pwsh
+        env:
+          ENCODED_VALUE: ${{ secrets.SECRING_GPG }}
+          OUPUT_PATH: .\secring.gpg
+      - name: Grant Execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Build with Gradle
+        run: ./gradlew build
+      - name: Upload a Build Artifact
```

Comment:
we probably need to add another task to upload the unit tests results since we don't have sonarqube on that repo.

Reply:
Where do the unit test results live in relation to the directory?

Reply:
`build/test-results` for the machine results (xml) and `build\reports\tests\test` for the human readable ones (html), we most likely only need the html one. Also the gradle task will fail when unit test fail, so that upload task needs to always run, even if a previous task as fail so we get the results of the tests no matter what.